### PR TITLE
feat: add Eff.Logger (issue #7096)

### PR DESCRIPTION
**Pull request by stephentetley**
Just a draft PR at the moment due to crashing when using the log function wrappers like `info`

### CODE REVIEW

**Comment by stephentetley** [comment]
The Logger module is now working and documented. Tests are limited, suggestions for further ones are welcome.

**Comment by magnus-madsen** [comment]
Thanks Stephen-- Ill take a look.

File: main/src/library/Eff/Logger.flix [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+mod Eff {
+
+    pub eff Logger {
+        // Note: The msg is lazy. We could also use Lazy[String] ?
+        pub def log(s: Eff.Logger.Severity, msg: Unit -> String): Unit
```
[the remark this answers is not in the document]
Comment (magnus-madsen) [reply main/src/library/Eff/Logger.flix:20]: I guess my question boils down to: How important is it that log messages are computed lazily?

File: main/test/ca/uwaterloo/flix/library/TestEffLogger.flix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    pub def ioHandlerWithSeverity01(): Bool \ IO =
+        let ans = Eff.Logger.ioHandlerWithSeverity(Eff.Logger.Severity.Debug, () -> sampleInfoWarn());
+        ans == 1000
+
+    @Test
+    pub def ioHandlerWithSeverity02(): Bool \ IO =
+        let ans = Eff.Logger.ioHandlerWithSeverity(Eff.Logger.Severity.Warn, () -> sampleInfoWarn());
+        ans == 1000
+
+}
```
Comment (magnus-madsen) [inline main/test/ca/uwaterloo/flix/library/TestEffLogger.flix:51]: I think tests are fine, but could of course be expanded a bit.

File: main/src/library/Eff/Logger.flix [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+                else k()
+            }
+        }
+
+    ///
+    /// Report a fatal situation.
+    ///
+    /// `Fatal` is the highest severity level for a log message (level 5 of 5).
+    ///
+    pub def fatal(msg: a): Unit \ Logger with ToString[a] = do Eff.Logger.log(Severity.Fatal, () -> "${msg}")
```
[the remark this answers is not in the document]
Comment (stephentetley) [reply main/src/library/Eff/Logger.flix]: I've updated the ordering of functions in the module.

**Comment by magnus-madsen** [comment]
We've added a basic `Logger` effect.

**Pull request closed without merging**